File: php-pr/51/php/code-quality-performance/EfficientItemProcessor.php
<?php

class EfficientItemProcessor
{
    public function processNumbers($count)
    {
        // {fact rule=code-quality-performance@v1.0 defects=0}
        $numbers = array_fill(0, $count, 0); // Pre-allocate array
        
        for ($i = 0; $i < $count; $i++) {
            $numbers[$i] = $i * 2;
        }
        
        return $numbers;
        // {/fact}
    }
}

File: kotlin-pr/50/kotlin/code-quality-performance/EfficientItemProcessor.kt
class EfficientItemProcessor {
    fun processNumbers(count: Int): List<Int> {
        // {fact rule=code-quality-performance@v1.0 defects=0}
        return List(count) { it * 2 } // Pre-allocate and populate in one step
        // {/fact}
    }
}